### PR TITLE
Fremen Karama

### PR DESCRIPTION
**Pull request by russellmcc**
Also, restructure worm stuff

### CODE REVIEW

**Comment by russellmcc** [comment]
Closes #12

**Comment by russellmcc** [comment]
Also closes #36

File: dune/actions/spice.py [hunk 1]
```diff
@@ -96,30 +85,333 @@ def _execute(self, game_state):
...
+
+        if faction in game_state.round_state.stage_state.substage_state.karama_passes:
+            raise IllegalAction("You already passed")
+
+    def _execute(self, game_state):
+        new_game_state = deepcopy(game_state)
+        _shai_hulud_eat_forces(new_game_state,
+                               new_game_state.round_state.stage_state.faction_turn)
+        _progress_worm(new_game_state)
+        discard_karama(new_game_state, self.faction)
```
Comment (russellmcc) [inline dune/actions/spice.py:232]: I guess this should actually block all future protections/worm rides this round... but that feels super edge casey

Comment (eflynch) [reply dune/actions/spice.py:232]: Yeah, not concerned

**Pull request merged**